coast/data/tidegauge.py

```diff
@@ -153,6 +187,100 @@ def read_gesla_v3(self, fn_gesla, date_start=None, date_end=None):
             self.dataset = dataset
             self.apply_config_mappings()
 
+    @classmethod
+    def _read_gesla_header_v5(cls, fn_gesla):
+        """
+        Reads header from a GESLA file (format version 5.0).
+
+        Parameters
+        ----------
+        fn_gesla (str) : path to gesla tide gauge file
+
+        Returns
+        -------
+        dictionary of attributes
+        """
+        debug(f'Reading GESLA header from "{fn_gesla}"')
+        fid = open(fn_gesla)
```

Comment:
You can use the with-block notation when opening files so that the file automatically closes itself after the code block is complete.

e.g.:
```python
with open(fn_gesla) as fid:
     # Read file and do stuff:
     format_version = float(fid.readline().split()[3])
     # Geographical stuff
     site_name = fid.readline().split()[3:]
     etc...
# Continue code outside of with block (With block has ended here so the file is closed automatically.)
print("example code")
```